### DISCUSSION TITLE
[Auth] Enhance app-locking options and encrypt local data

### DISCUSSION THREAD

**Discussion by borque** [ente-io/ente#1874] [Enhancements]
Hello!
Thanks for Ente auth! I would like to suggest to enhance the locking of Ente auth (to a more Bitwarden-like style):
1) Offer to create an app-password, that can be different from the PIN needed to unlock the device an still is not the main encryption key.
2) Offer to select a delay before the app is relocked while it is still open in background.

Edit:
Meanwhile I have installed the desktop apps and found that the do not offer any kind of locking beyond the OS' locking screen and that all local data seems to be stored unencrypted in a json file in the home directory.
So I would like to extend my suggestion: Encrypt the locally stored data.

I think Bitwardens approach to locking and encrypting is very reasonable and good for both, mobile and desktop apps.

Thanks and greetings!

**Comment by cbmiller2610** [comment]
This would be great, TouchID lock is a limited form of security and I would prefer a separate PIN!

**Comment by daithi-coyle** [comment]
+1 to this. Current behaviour on Windows is to open the app, unlocked with all 2fa visible upon login. Seems very insecure.
Windows Hello integration (fingerprint) would be ideal but even a pin would be better than nothing.

**Comment by r4ld** [comment]
+1. On iOS, it currently requires TouchID, but if TouchID failed, it will default to device passcode. Now if both my device and device's passcode are compromised, they may gain access to my device as well as this app. That's not to say if my device doesn't have passcode set up, the app will not be locked too.

**Comment by gitmeED331** [comment]
For those on linux, we are given nothing to secure the app or the entries.
We need a PIN at the very least. 
Biometric would be useful and convenient as well. 

The current state of the linux app is horrible for security. On top of no way to secure the app, there is no "start-in-tray" option which then results in showcasing all your accounts with no prevention of view secrets or preventing editing.

And I just realized that if I click the "passkeys" in the app, it opens up the add passkey page in a web browser. No pin or password required. So anyone that gains access to my computer before it locks or breaks into it, they can do even greater damage. 

Local encryption of data should be a no brainer first stop. An Authentication app is a security app, which means everything needs to be encrypted, even locally.

From a security standpoint, the Ente Auth desktop apps are terrible.

Sure, there is the "lockscreen" option in the settings, I have no idea what it does since it doesn't recognize my lockscreen (which is built in Aylur's GTK Shell and purely custom, but it does support PAM and 2FA).

OS: Arch Linux
Desktop: Hyprland
Shell: AGS

**Comment by kirbw** [comment]
+1 on the ability to auto-lock Auth on the desktop after a period of time.

**Comment by ashilkn** [comment]
Custom lock (PIN or password) for Auth has been shipped in the [latest release](https://github.com/ente-io/ente/releases/tag/auth-v3.1.3)

**Comment by m4to-3pe** [comment]
This issue does not seem to have been resolved.* While it is trivial to create a script to manually implement a "timeout", aka wait 30 sec and then pkill the app, it looks like information is still stored in ~/.local/share/io.ente_auth/shared_preferences.json. It can be cleared manually (and should be, in case you want to uninstall the app), but then obviously there is no persistant login.

I'm very far from a security researcher, so to me the info stored in that .json looks rather insecure. Did https://github.com/ente-io/ente/discussions/1874#discussioncomment-10415285, so [the above mentioned release](https://github.com/ente-io/ente/releases/tag/auth-v3.1.3) or a later one introduce encryption (at rest) while locked?

Edit:
*Atleast with the AppImage, there is still no way to natively do suggestion (2).
Also to me this feels to fit rather in the security and not in the enhancement category (albeit probably a low or moderate severity issue at most).
